### DISCUSSION TITLE
Aspire service discovery support

### DISCUSSION THREAD

**Discussion by petermorlion** [reactiveui/refit#1940] [General]
For a course on architecture, I want to use Refit to showcase N-tier architectures. But I can't seem to get Refit to work with Aspire service discovery. Basically, what I have is this what I have in my Aspire AppHost:

```
var businessTier = builder.AddProject<Projects.BusinessTier>("businesstier")
    .WithReference(db)
    .WaitFor(db);

builder.AddProject<Projects.PresentationTier>("presentationtier")
    .WithReference(businessTier)
    .WaitFor(businessTier);
```

In my Presentation tier, the Program.cs has this:

```
builder.Services.AddServiceDiscovery(); // doesn't work with or without this
builder.Services.AddRefitClient<IBusinessLayerApi>()
    .ConfigureHttpClient(c => c.BaseAddress = new("https://businesstier"));
```

(where `IBusinessLayerApi` is the interface where I put Refit's attributes on)

From the [official docs](https://learn.microsoft.com/en-us/dotnet/aspire/service-discovery/overview) I assumed that something 🪄magical🪄 was going to happen and that the URL would be replaced with the correct one (https://localhost:7195 in my case), but that doesn't seem to happen?

What's the correct way to integrate Refit with Aspire?

**Comment by Suiram1701** [comment]
Normally this should work as expected and at least for me it works

> 🪄magical🪄 was going to happen

Resolving the service url from its name is done by the HttpClient you get from IHttpClientFactory. At least today (idk if it already were done like this when you tried) Refit uses internally a client provided by this factory so this should work as expected. 

If this probleme still exist could you please provide the steps to reproduce this issue?

**Reply by petermorlion** [reply]
I created a small test and indeed it works now. I don't know what I did wrong exactly, but it works. This issue can be closed.
